### DISCUSSION TITLE
Do `x-on` handlers qualify for "user gesture event handler"?

### DISCUSSION THREAD

**Discussion by northern-mindset** [alpinejs/alpine#3508] [1. Help] [answered]
In https://caniuse.com/mdn-api_clipboard_writetext we read that for Safari this API:

> Must be called within user gesture event handlers such as `pointerdown` or `pointerup`.

Question: if we call this API in the `x-on:click` event handler, does it fall into the mentioned category of “user gesture event handlers?” In other words, doesn't Alpine magic somehow execute the handler outside the context of user gesture?

Examining the call stack in the browser didn't help me figure this out.

**Comment by ekwoka** [comment, marked as answer]
Yes. It should work fine, no Alpine won't call it in another context except if you did something that resulted in the event queue being used to resolve to it.

What happens before the clipboard write?

**Reply by northern-mindset** [reply]
> What happens before the clipboard write?

Not sure I understand what you are asking about here. Nevertheless, a clean experiment showed that it really works fine. I was just confused when in one experiment in BrowserStack I got a deny error, though it was late in the night, and most probably I messed up something. That is why I asked the question, to get the “theory”.

Anyhow, it's good to have it documented here. Thank you for the confirmation.

**Reply by ekwoka** [reply]
I meant in the event handler, what happens before you try to write to the keyboard?

If you awaited something, the context could change, if you use some kind of delayed callback, the context could change.

**Reply by northern-mindset** [reply]
Oh, I see. Makes sense. In my case there is none of these circumstances, hence it works. Thanks again.
